src/main/java/mate/academy/AsyncRequestProcessor.java

```diff
@@ -1,16 +1,29 @@
 package mate.academy;
 
+import java.util.Map;
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.Executor;
+import java.util.concurrent.TimeUnit;
 
 public class AsyncRequestProcessor {
     private final Executor executor;
+    private final Map<String, UserData> cache = new ConcurrentHashMap<>();
 
     public AsyncRequestProcessor(Executor executor) {
         this.executor = executor;
     }
 
     public CompletableFuture<UserData> processRequest(String userId) {
-        return null;
+        return CompletableFuture.supplyAsync(() ->
+                cache.computeIfAbsent(userId, id -> {
+                    try {
+                        // Simulate a delay for fetching user data
+                        TimeUnit.SECONDS.sleep(1);
+                    } catch (InterruptedException e) {
+                        Thread.currentThread().interrupt();
+                    }
```

Comment:
Catching InterruptedException and just resetting the thread's interrupt flag is not enough to handle the interrupted exception properly. You should also consider how the interruption should affect the current operation and the data being returned. If the thread was interrupted while sleeping, it might be that the operation should not continue. You should either throw an exception or return a special value indicating that the operation was cancelled.